zk_prover/src/merkle_sum_tree/entry.rs

```diff
@@ -32,22 +32,28 @@ impl<const N_ASSETS: usize> Entry<N_ASSETS> {
     }
 
     pub fn compute_leaf(&self) -> Node<N_ASSETS>
+    where
+        [usize; N_ASSETS + 1]: Sized,
+    {
+        self.create_node(&self.balances)
```

Comment:
As you added the api `middle` on the struct `Node`, can you also add an api `leaf` that replaces the function `create_node` on `Entry`?